src/org/zaproxy/zap/extension/websocket/manualsend/WebSocketPanelSender.java

```diff
@@ -80,14 +86,59 @@ public void cleanup() {
     }
     
     private WebSocketProxy getDelegate(Integer channelId) {
-    	if (!connectedProxies.containsKey(channelId)) {
+    	if (!connectedProxies.containsKey(channelId) && !closedProxies.containsKey(channelId)) {
     		logger.warn("Selected WebSocket channel is not connected. Unable to send manual crafted message!");
     		throw new IllegalArgumentException(Constant.messages.getString("websocket.manual_send.fail.disconnected_channel") 
     				+ " " + Constant.messages.getString("websocket.manual_send.fail"));
-    	}
+    	}else if (closedProxies.containsKey(channelId)){
+			logger.warn("Selected WebSocket channel is not connected. Unable to send manual crafted message!");

```

Comment:
This doesn't check that it's not connected, it only checks that it's in the closed list....

Reply:
I am not sure what is the meaning of that. Every channel has two states either closed or opened. Whenever channel's status is changed, channel is added to the appropriate HashMap.